### PR TITLE
ORTB 2.6: Harden SetImps to cover larger changes

### PR DESCRIPTION
**Pull request by hhhjort**

### CODE REVIEW

**Review by VeronikaSolovei9 — APPROVED** [review]

**Comment by bsardo** [comment]
LGTM but should we add a unit test for `SetImp` that ensures the same number of elements appear in `Imp` and the imp wrappers and that they reference the same addresses via `assert.Same`?

File: openrtb_ext/request_wrapper.go [hunk 1]
```diff
@@ -95,6 +95,12 @@ func (rw *RequestWrapper) GetImp() []*ImpWrapper {
 
 func (rw *RequestWrapper) SetImp(imps []*ImpWrapper) {
 	rw.impWrappers = imps
+	imparr := make([]openrtb2.Imp, len(imps), len(imps))
```
Comment (SyntaxNode) [inline openrtb_ext/request_wrapper.go:98]: Nitpick: There is not need to specify both a length and capacity when they are the same. Just a single term will suffice here.

File: openrtb_ext/request_wrapper.go [hunk 2]
```diff
@@ -95,6 +95,12 @@ func (rw *RequestWrapper) GetImp() []*ImpWrapper {
...
+	for i, imp := range imps {
```
Comment (SyntaxNode) [inline openrtb_ext/request_wrapper.go:99]: Nitpick: `imp` is actually an `impWrapper`.

File: openrtb_ext/request_wrapper_test.go [hunk 3]
```diff
@@ -1884,6 +1884,37 @@ func TestImpWrapperGetImpExt(t *testing.T) {
...
+	*iw[2] = ImpWrapper{Imp: &expectedImps[2]}
+	iw = append(iw, &ImpWrapper{Imp: &expectedImps[3]})
+
+	rw.SetImp(iw)
+	assert.Equal(t, expectedImps, rw.BidRequest.Imp)
+	iw = rw.GetImp()
+	// Ensure that the wrapper pointers are in sync.
+	for i, _ := range rw.BidRequest.Imp {
+		// Assert the pointers are in sync.
+		assert.Equal(t, &rw.Imp[i], iw[i].Imp)
```
Comment (SyntaxNode) [inline openrtb_ext/request_wrapper_test.go:1913]: To test for pointer equality, you want to use `assert.Same` to compare the pointer values. This tests the values pointed and doesn't seem to be what you are looking for here based on the description.

**Review by bsardo — APPROVED** [review]

File: openrtb_ext/request_wrapper.go (hunk 1 above)
Comment (hhhjort) [reply openrtb_ext/request_wrapper.go]: Thanks

File: openrtb_ext/request_wrapper_test.go (hunk 3 above)
Comment (hhhjort) [reply openrtb_ext/request_wrapper_test.go]: Thanks

File: openrtb_ext/request_wrapper.go (hunk 2 above)
Comment (hhhjort) [reply openrtb_ext/request_wrapper.go]: Thanks

**Review by bsardo — APPROVED** [review]

**Pull request merged**